File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/CityUseCase.java
package com.kazakago.cleanarchitecture.domain.usecase;

import com.kazakago.cleanarchitecture.domain.model.city.CityModel;

import java.io.IOException;

import io.reactivex.Observable;

/**
 * City UseCase
 *
 * @author Kensuke
 */
public interface CityUseCase {

    Observable<CityModel> findAll() throws IOException;

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/CityUseCase.kt
package com.kazakago.cleanarchitecture.domain.usecase

import com.kazakago.cleanarchitecture.domain.model.city.CityModel
import io.reactivex.Observable
import java.io.IOException

/**
 * City UseCase
 *
 * @author Kensuke
 */
interface CityUseCase {

    @Throws(IOException::class)
    fun findAll(): Observable<CityModel>

}